pykeepass/kdbx_parsing/common.py

```diff
@@ -128,6 +128,7 @@ def compute_key_composite(password=None, keyfile=None):
     # hash the keyfile
     if keyfile:
         if hasattr(keyfile, "read"):
+            keyfile.seek(0)
```

Comment:
This is a bit dangerous, you need to check

- That there is a seek method
- That [seekable](https://docs.python.org/3/library/io.html#io.IOBase.seekable)() returns True (and that the method exists)

see https://docs.python.org/3/library/io.html#class-hierarchy